zoo/services/migrations/0014_add_environment.py

```diff
@@ -0,0 +1,87 @@
+# Generated by Django 2.1.7 on 2019-10-11 12:44
+
+import django.contrib.postgres.fields
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def move_data(apps, schema_editor):
+    Service = apps.get_model("services", "Service")
+    Environment = apps.get_model("services", "Environment")
+
+    for service in Service.objects.all():
+        Environment.objects.create(
+            service_id=service.id,
+            name="default",
+            service_urls=[service.service_url] if service.service_url else [],
+            health_check_url=service.health_check_url,
+            dashboard_url=service.dashboard_url,
+        )
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [("services", "0013_auto_20190515_1451")]
+
+    operations = [
+        migrations.CreateModel(
+            name="Environment",
+            fields=[
+                (
+                    "id",
+                    models.AutoField(
+                        auto_created=True,
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                ("name", models.CharField(max_length=100)),
+                (
+                    "service_urls",
+                    django.contrib.postgres.fields.ArrayField(
+                        base_field=models.URLField(max_length=500),
+                        default=list,
+                        null=True,
+                        size=None,
+                    ),
+                ),
+                (
+                    "health_check_url",
+                    models.URLField(blank=True, max_length=500, null=True),
+                ),
+                (
+                    "dashboard_url",
+                    models.URLField(blank=True, max_length=500, null=True),
+                ),
+            ],
+        ),
+        migrations.AddField(
+            model_name="environment",
+            name="service",
+            field=models.ForeignKey(
+                on_delete=django.db.models.deletion.CASCADE,
+                related_name="environments",
+                related_query_name="environment",
+                to="services.Service",
+            ),
+        ),
+        migrations.AlterUniqueTogether(
+            name="environment", unique_together={("service", "name")}
+        ),
+        migrations.RunPython(move_data),
```

Comment:
I'd also add a reverse migration, just in case we have to roll it back 🙂

Reply:
Good idea :-)